src/tools/smokeVisionProcess.ts

```diff
@@ -37,6 +37,9 @@ async function ComputeShadow(eventDetail: Detail)
         return;
     }
 
+    // remove debug visualisations from any previous pass..
+    OBR.scene.local.deleteItems((await OBR.scene.local.getItems(f => f.metadata[`${Constants.EXTENSIONID}/debug`] === true)).map(i => i.id));
```

Comment:
This needs to be awaited, or else it could fire, the rest of the code will keep going, and it'll finish somewhere down the line.

Reply:
I left it in because I use it for debugging visualisations during development. It doesn't matter for users. I'll remove it and consider adding a Dev debugging flag instead